modularized polyfills for web5 and made web5 check on Docusaurus

dayhaysoos opened the pull request:
Introduced a number of changes here:

- Removed polyfill from checkWeb5 function as it's not necessary in all environments
- Add testing script to root level of pnpm
- updated node based environments with polyfill

leordev commented on package.json:
```diff
@@ -2,5 +2,9 @@
     "@web5/api": "0.8.1"
+  },
+  "scripts": {
+    "test:web5": "pnpm -r --workspace-concurrency 1 --no-bail test:web5",
+    "test:dwn": "pnpm -r --workspace-concurrency 1 --no-bail test:dwn"
```
Could you please update our pipeline workflow file to simplify the tests commands to use these ones? they should be just `pnpm test:X`

dayhaysoos replied:
I'm actually confused by this. Do you want me to change these commands to just be `pnpm test:X` or do you want to call these commands by doing `pnpm test:X` from the root level?

leordev replied:
Oh apologies for the confusion. Just simplify these calls:
https://github.com/TBD54566975/bundler-bonanza/blob/main/.github/workflows/tests-runner.yml#L58
https://github.com/TBD54566975/bundler-bonanza/blob/main/.github/workflows/tests-runner.yml#L94

leordev commented on tests/docusaurus/src/pages/index.js:
```diff
@@ -5,32 +5,36 @@ import useDocusaurusContext from "@docusaurus/useDocusaurusContext";
     >
       <main>Web5 stuff right here</main>
-      {web5 && <div>{web5.did}</div>}
+      {web5 && <div>{web5}</div>}
```
beautiful. now you only need to check if this div is rendered in a browser test framework right? (I think that could be the next pr!)

leordev replied:
(I think you should also do the same for the dwn test)

dayhaysoos replied:
Yes, for the dev site we use vitest for doing this (but we'll talk about it)

leordev replied:
From [vitest browser docs](https://vitest.dev/guide/browser.html#different-ways-of-testing):
> [jsdom](https://www.npmjs.com/package/jsdom) is an example of a spec implementation that simulates a browser environment by being used with a test runner like Jest or Vitest, while other testing tools such as [WebdriverIO](https://webdriver.io/) or [Cypress](https://www.cypress.io/) allow developers to test their applications in a real browser or in case of [Playwright](https://playwright.dev/) provide you a browser engine.

I think here we should use a real browser/browser engine to be as much close to reality as possible! And based in our talks before, my vote goes for playwright!

Quick start: https://playwright.dev/docs/intro
Adding to our CI: https://playwright.dev/docs/ci-intro

leordev commented on tests/docusaurus/src/pages/index.js:
```diff
@@ -5,32 +5,36 @@ import useDocusaurusContext from "@docusaurus/useDocusaurusContext";
-      } catch (error) {
-        console.error("Failed to connect to Web5:", error);
-      }
+      const result = await checkWeb5(Web5);
+
+      setWeb5(result.did.connectedDid);
```
(I think you should also do the same for the dwn test)

leordev commented on tests/util/dwn-test.js:
```diff
@@ -1,4 +1,4 @@
-require("./webcrypto-polyfill.js");
+require("./node-polyfill.js");
```
you should remove this line, for the same reason that we did it in the `web5-test.js`. You will stumble on it when you test dwn in docusaurus

dayhaysoos commented:
@leordev I just made a bunch of changes:

- Added webpack polyfill to support Dwn tests
- Adjusted check dwn to return results so we can render them on the browser
- polyfilled buffer as well at the top of checkDwn function, not sure if this was okay. Could this introduce more issues?

leordev commented on tests/util/dwn-test.js:
```diff
@@ -1,4 +1,4 @@
-require("./webcrypto-polyfill.js");
+global.Buffer = global.Buffer || require("buffer").Buffer;
```
if this is specifically for docusaurus I'd add it in the proper docusaurus file for it, analogous to what we did for the node-cjs and node-esm above to import the polyfills since they are envs that require it!

dayhaysoos replied:
I agree, but I actually don't know how to do it. I've tried importing Buffer directly, I tried adding it to the webpack polyfill as well. It always gives Buffer can't be found error. Am I missing something?
